compose/project.py

```diff
@@ -205,6 +215,27 @@ def kill(self, service_names=None, **options):
     def remove_stopped(self, service_names=None, **options):
         parallel.parallel_remove(self.containers(service_names, stopped=True), options)
 
+    def initialize_volumes(self):
+        try:
+            for volume in self.volumes:
+                volume.create()
+        except NotFound:
+            raise ConfigurationError(
+                'Volume %s specifies nonexistent driver %s' % (volume.name, volume.driver)
+            )
+        except APIError as e:
+            if 'Choose a different volume name' in str(e):
```

Comment:
This is really unfortunate. The volumes API should really be returning a proper 4xx code for this. It's a shame we have to resort to string comparison for this.